### PR TITLE
Remove deadcode

### PR DESCRIPTION
**Pull request by 100mik**
Removing deadcode detected by `golangci-lint` on [Sonatype Lift](https://lift.sonatype.com/result/longhorn/longhorn-manager/01FHW2JKTWJ3Z4RA5MMT2A53WS?t=CustomTool%20%22golangci-lint%22%7Cdeadcode).

### CODE REVIEW

**Review by shuo-wu — APPROVED** [review]

File: app/driver.go [hunk 1]
```diff
@@ -319,16 +319,6 @@ type DaemonSetOps struct {
 	kubeClient *clientset.Clientset
 }
 
-func newDaemonSetOps(kubeClient *clientset.Clientset) (*DaemonSetOps, error) {
```
Comment (shuo-wu) [inline app/driver.go:322]: After rechecking the code, it seems that struct `DaemonSetOps` is not used as well. I think you can remove the struct as well as all methods/functions for it.

File: client/generated_node_input.go [hunk 2]
```diff
@@ -28,12 +28,6 @@ type NodeInputOperations interface {
 	Delete(container *NodeInput) error
 }
 
-func newNodeInputClient(rancherClient *RancherClient) *NodeInputClient {
```
Comment (shuo-wu) [inline client/generated_node_input.go:31]: Similar here. It seems that this whole go file is useless.

File: types/types.go [hunk 3]
```diff
@@ -170,7 +170,6 @@ const (
 
 	BackingImageDataSourcePodNamePrefix = "backing-image-ds-"
 
-	shareManagerPrefix    = "share-manager-"
```
[the remark this answers is not in the document]
Comment (100mik) [reply types/types.go:173]: made this change!

File: app/driver.go [hunk 4]
```diff
@@ -313,6 +315,7 @@ func deployCSIDriver(kubeClient *clientset.Clientset, lhClient *lhclientset.Clie
 
 	return nil
 }
+<<<<<<< HEAD
```
Comment (shuo-wu) [inline app/driver.go:318]: Need to remove these lines.

Comment (100mik) [reply app/driver.go]: My bad missed this. Done!

File: types/types.go [hunk 5]
```diff
@@ -414,11 +415,11 @@ func GetShareManagerImageChecksumName(image string) string {
 }
 
 func GetShareManagerPodNameFromShareManagerName(smName string) string {
-	return LonghornLabelShareManager + "-" + smName
+	return shareManagerPrefix + "-" + smName
```
Comment (joshimoo) [inline types/types.go:418]: Since the prefix already includes, the `-` sign, the appending of the `-` is not longer needed.

File: types/types.go [hunk 6]
```diff
@@ -414,11 +415,11 @@ func GetShareManagerImageChecksumName(image string) string {
...
 }
 
 func GetShareManagerNameFromShareManagerPodName(podName string) string {
-	return strings.TrimPrefix(podName, LonghornLabelShareManager+"-")
+	return strings.TrimPrefix(podName, shareManagerPrefix+"-")
```
Comment (joshimoo) [inline types/types.go:422]: Since the prefix already includes, the - sign, the appending of the - is not longer needed.

File: types/types.go [hunk 7]
```diff
@@ -170,7 +170,8 @@ const (
 
 	BackingImageDataSourcePodNamePrefix = "backing-image-ds-"
 
-	shareManagerPrefix    = "share-manager-"
+	shareManagerPrefix = "share-manager-"
+
```
Comment (joshimoo) [inline types/types.go:174]: NIT: no need for a separate section for this variable.

**Review by joshimoo — CHANGES_REQUESTED** [review]
LGTM, when replacing the usage with the const var, the extra `-` was overlooked. After that is fixed this can be merged.

File: types/types.go (hunk 7 above)
Comment (100mik) [reply types/types.go]: Incorporated suggestions!

**Review by joshimoo — APPROVED** [review]
LGTM

**Pull request merged**